src/main/java/net/javadiscord/javabot/systems/help/AutoCodeFormatter.java

```diff
@@ -0,0 +1,102 @@
+package net.javadiscord.javabot.systems.help;
+
+import lombok.AllArgsConstructor;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.javadiscord.javabot.data.config.BotConfig;
+import net.javadiscord.javabot.systems.moderation.AutoMod;
+import net.javadiscord.javabot.systems.user_preferences.UserPreferenceService;
+import net.javadiscord.javabot.systems.user_preferences.model.Preference;
+import net.javadiscord.javabot.util.ExceptionLogger;
+import net.javadiscord.javabot.util.WebhookUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+
+/**
+ * Handles auto-formatting of code in help-channels.
+ * A non-listener class since it has to be called strictly after {@link HelpListener} to not send its messages before channel reserving.
+ */
+@AllArgsConstructor
+public class AutoCodeFormatter {
+	private final AutoMod autoMod;
+	private final BotConfig botConfig;
+	private final UserPreferenceService preferenceService;
+
+	/**
+	 * Method responsible for finding a codeblock, if present.
+	 *
+	 * @param event a {@link MessageReceivedEvent}.
+	 * @return a MessageCodeblock instance, holding a startIndex, content and an endIndex
+	 */
+	@Nullable
+	private static AutoCodeFormatter.CodeBlock findCodeblock(@NotNull MessageReceivedEvent event) {
+		String msg = event.getMessage().getContentRaw();
+		int openingBracket = msg.indexOf("{");
+		int closingBracket = msg.lastIndexOf("}");
+		if (closingBracket == -1 || openingBracket == -1) return null;
+		int startIndex = msg.lastIndexOf("\n", openingBracket);
+		int endIndex = msg.indexOf("\n", closingBracket);
+		if (startIndex == -1) startIndex = 0;
+		if (endIndex == -1) endIndex = msg.length();
+		return new CodeBlock(startIndex, endIndex);
+	}
+
+	/**
+	 * called by {@link HelpListener#onMessageReceived(MessageReceivedEvent)} on every message.
+	 * It is worth noting that this class can't register as an event handler itself due to there being no way of
+	 * setting its methods to be strictly called after {@link HelpListener}.
+	 *
+	 * @param event a {@link MessageReceivedEvent}
+	 */
+	public void handleMessageEvent(@Nonnull MessageReceivedEvent event) {
```

Comment:
I guess that (or the whole class) could be package private since it's just for the help system.